build/index.js

```diff
@@ -137,13 +137,74 @@ var ReactMultiSelect = function (_React$Component) {
         _this.state = {
             dropDownClicked: false
         };
+
+        _this.buildDropDown = _this.buildDropDown.bind(_this);
+        _this.onSelectAll = _this.onSelectAll.bind(_this);
+        _this.onSelectNone = _this.onSelectNone.bind(_this);
         return _this;
     }
 
     _createClass(ReactMultiSelect, [{
+        key: "buildDropDown",
+        value: function buildDropDown() {
+            var _this2 = this;
+
+            var optionsList = [];
+            var optionsListStyles = {};
+            var selectedCount = 0;
+
+            optionsList = this.props.options.map(function (el, i) {
+                if (el.value) {
+                    selectedCount++;
+                    optionsListStyles = _this2.props.optionsListStyles;
+                } else {
+                    optionsListStyles = {};
+                }
+                return _react2.default.createElement(
+                    "li",
+                    { key: el.id, value: el.value },
+                    _react2.default.createElement(
+                        "div",
+                        { className: "option-list", style: _this2.props.disabled ? disabledStyle : optionsListStyles, onClick: _this2.optionsOnchange.bind(_this2, i, !el.value) },
+                        el.label
+                    )
+                );
+            });
+
+            if (this.props.enableSelectAllorNone) {
```

Comment:
If you are using single select and enableSelectAllorNone={true}, the component basic behavior of single select changes.
Can you add some more condition for avoiding this.

Reply:
Appropriate your effort for this feature. Planing to do some more thing in this, will be glad to collaborate with you.

Reply:
Hi,
Sorry for the delayed response, was on Diwali vacation.
I will make the required changes.

Thanks
Arun

Reply:
Fixed and Pushed a new commit